debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -532,10 +532,15 @@ protected SourceRecord replaceOffset(SourceRecord record) {
     }
 
     protected void enqueueSchemaChanges(String dbName, String ddlStatements) {
-        if (context.includeSchemaChangeRecords() &&
-                context.makeRecord().schemaChanges(dbName, ddlStatements, super::enqueueRecord) > 0) {
-            logger.debug("Recorded DDL statements for database '{}': {}", dbName, ddlStatements);
+        if (!context.includeSchemaChangeRecords() || ddlStatements.length() == 0) {
+            return;
         }
+        logger.info("Recorded the following DDL statements for database '{}':", dbName);
+        Arrays.stream(ddlStatements.split(";" + System.lineSeparator())).forEach(ddlStatement -> {
```

Comment:
@hchiorean, I know this is what we talked about yesterday, but this splitting on `;` concerns me because if any of the statements were to contain a `;` character, then the split logic would break that one statement into several.

Perhaps an alternative is to not join but instead of [one call to `schema.applyDdl` with a list on line 295](https://github.com/debezium/debezium/pull/84/files#diff-ed5ffc9991fc93f25a27c0b78d98f724R295) we call `schema.applyDdl` for each of the statements in the list. Or, @mvj3's [suggestion](https://github.com/DataPipelineInc/debezium/commit/3b9dddb51fd461d4e5b63fd5453d68cecf92cf91) doesn't even add the statements to the list but instead calls `schema.applyDdl` for each statement.

Thoughts?